asmrun/amd64.S

```diff
@@ -343,13 +336,20 @@ G(caml_system__code_begin):
 #define DW_CFA_val_expression     0x16
 #define DW_CFA_expression         0x10
 #define DW_CFA_def_cfa_expression 0x0f
+#define DW_CFA_restore            0x03
+#define DW_CFA_val_offset         0x14
+#define DW_REG_rbp                6
+#define DW_REG_rsp                7
+#define DW_REG_retaddr            16
+#define DW_OP_breg                0x70
```

Comment:
Which register is this on amd64?

Reply:
The 'retaddr' reg is a fiction of DWARF. You specify the return address of the next frame as the contents of a pseudoregister (or, on e.g. POWER, an actual register). The standard definition of the 'retaddr' reg on x86_64 is that it's the fictional register 16 that was "spilled" just after the CFA.

I'll delete this, it's not actually needed at the moment. An earlier version of the DWARF code had DWARF expressions to find the previous stack pointer and return address on a different stack, but it's easier to do the traversal for the CFA, and leave the CFA -> retaddr and CFA -> stack computations as standard.